osquery/core/database/rocksdb_database.h

```diff
@@ -0,0 +1,106 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <unordered_map>
+
+#include <boost/filesystem.hpp>
+#include <rocksdb/db.h>
+
+#include <osquery/core/database/database.h>
+#include <osquery/database/plugins/rocksdb.h>
+
+namespace osquery {
+
+enum class RocksdbError {
+  UnexpectedValueType = 1,
+  DatabaseIsCorrupted = 2,
+  BatchWriteFail = 3,
+};
+
+// This implementaion is not 100% thread safe and does not
+// support close/open commands during regular db usage.
+// It's safe as long as DB is open in single threaded manner,
+// and only then used from multiple threads.
+class RocksdbDatabase final : public Database {
+ public:
+  using Handle = rocksdb::ColumnFamilyHandle;
+  using HandleRef = std::shared_ptr<Handle>;
+
+  RocksdbDatabase(std::string name) = delete;
+  explicit RocksdbDatabase(std::string name, std::string path)
+      : Database(std::move(name)), path_(std::move(path)){};
+  ~RocksdbDatabase() override {}
+
+  ExpectedSuccess<DatabaseError> destroyDB() override;
+  ExpectedSuccess<DatabaseError> open() override;
+
+  // Open and close are not thread safe commands,
+  // Before closing db you need to ensure that no other theads are currenly
+  // using db connection.
+  // Please see comment above about reopening db.
+  void close() override;
+
+  Expected<int32_t, DatabaseError> getInt32(const std::string& domain,
```

Comment:
As far as `int32_t` is only one numeric type that is allowed to be stored to db would you like to make an alias to this type like `using Database::Int = int32_t;` and use it everywhere?

Reply:
In future db will likely support more types including i64 and i8